### DISCUSSION TITLE
Seeking clarification on the distinction between Points, Nodes, Vertices, and Quadrature Points in libmesh

### DISCUSSION THREAD

**Discussion by hityyds** [idaholab/moose#26837] [Q&A General] [answered]
I am working on calculating the center point of an element's side by averaging the coordinates of each vertex. I have been exploring the information available regarding libmesh::Elem for this purpose. However, I am currently confused about the differences among Point, Node, Vertex, and Quadrature Point.

My current understanding is as follows:
    A Point solely contains coordinate information.
    A Node builds upon a Point and is associated with degrees of freedom; every Node contains a Point.
    In MOOSE, each Quadrature Point corresponds to unknown variables (i.e. degrees of freedom), suggesting it should be linked to a Node.
    The number of Vertices is determined by the type of the element, representing physically existing points.

The confusion arises from the fact that both Nodes and Vertices seem to use the same indexing scheme when accessing their positions via the `elem->point(n)` function. 
```
 Point Elem::vertex_average() const
 {
   Point cp;
 
   const auto n_vertices = this->n_vertices();
 
   for (unsigned int n=0; n<n_vertices; n++)
     cp.add (this->point(n));
 
   return (cp /= static_cast<Real>(n_vertices));
 }
std::unique_ptr<MeshBase>
CoarseMeshMappingForCMFD::generate()
{
    ......

    // Check if the fine elem is nested in the coarse element
    for (unsigned int n = 0; n < elem->n_nodes(); n++)
    {
      // Get the node: we need to manually move it towards the centroid to
      // ensure that nothing weird happes due to round-off
      Point current_node = elem->point(n);

      ......
}
```
This gives me the impression that Vertices and Nodes share the same index set, which seems contradictory given that their counts are different.

Furthermore, my understanding suggests that each element has its unique set of Quadrature Points while Nodes can be shared across elements. This seemingly contradicts my earlier notion that each Quadrature Point corresponds to a single Node.

I would greatly appreciate your help in clarifying these concepts and how they relate to finding the center point of an element's side. Thank you in advance for your attention and assistance.

**Comment by rnpratoori** [comment, marked as answer]
You are wrong to conclude that each Quadrature point corresponds to a node. Quadrature point is usually something that is located within the element, not on the boundary/vertices of the element like nodes. Quadrature points differ based on the shape of elements and more predominantly the shape functions used. Both of these will determine how many Quadrature points are there for a particular element and also their respective locations.

**Reply by hityyds** [reply]
Thank you very much for your reminder.
